showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITUnaryDeadline.java

```diff
@@ -0,0 +1,448 @@
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.retrying.RetryingFuture;
+import com.google.api.gax.rpc.DeadlineExceededException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.showcase.v1beta1.BlockRequest;
+import com.google.showcase.v1beta1.BlockResponse;
+import com.google.showcase.v1beta1.EchoClient;
+import com.google.showcase.v1beta1.EchoSettings;
+import com.google.showcase.v1beta1.stub.EchoStubSettings;
+import com.google.showcase.v1beta1.stub.SequenceServiceStubSettings;
+import io.grpc.ManagedChannelBuilder;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.Test;
+import org.threeten.bp.Duration;
+
+/**
+ * For this test, we test a combination of various RetrySettings to try and ensure that the timeouts
+ * set by the customer are cancelled when timeouts have exceeded their limits
+ *
+ * <p>Each test attempts to get the number of attempts done in each call. The attemptCount is
+ * incremented by 1 as the first attempt is zero indexed.
+ */
+public class ITUnaryDeadline {
+
+  @Test
+  public void testGRPC_unarySuccessfulResponse_NoDeadlineExceeded()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    RetrySettings defaultNoRetrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRpcTimeout(Duration.ofMillis(5000L))
+            .setRpcTimeoutMultiplier(1.0)
+            .setMaxRpcTimeout(Duration.ofMillis(5000L))
+            .setTotalTimeout(Duration.ofMillis(5000L))
+            // Explicitly set retries as disabled (maxAttempts == 1)
+            .setMaxAttempts(1)
+            .build();
+    EchoStubSettings.Builder grpcEchoSettingsBuilder = EchoStubSettings.newBuilder();
+    grpcEchoSettingsBuilder.blockSettings().setRetrySettings(defaultNoRetrySettings);
+    EchoSettings grpcEchoSettings = EchoSettings.create(grpcEchoSettingsBuilder.build());
+    grpcEchoSettings =
+        grpcEchoSettings
+            .toBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                EchoSettings.defaultGrpcTransportProviderBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .build();
+    try (EchoClient grpcClient = EchoClient.create(grpcEchoSettings)) {
+      BlockRequest blockRequest =
+          BlockRequest.newBuilder()
+              .setSuccess(BlockResponse.newBuilder().setContent("gRPCBlockContent_3sDelay_noRetry"))
+              .setResponseDelay(com.google.protobuf.Duration.newBuilder().setSeconds(3).build())
+              .build();
+      RetryingFuture<BlockResponse> retryingFuture =
```

Comment:
Is this how customers would make use of retrys? If it is, we need to document it in the retry guide as well. I might be wrong but I thought we just make a unary call as usual and the retries would happen behind the scene.

Reply:
It's not how customers should be using these callables by default. I've only exposed the RetryingFuture myself since that's the only way I could track the number of attempts. I'll add it to the guide as a way to get the number of attempts.